File: NOVEMBER-2025/27-11-2025/3381.dart
class Solution {
  int maxSubarraySum(List<int> nums, int k) {
    List<int> minPrefix = List.filled(k, 1 << 60);
    minPrefix[0] = 0;

    int prefix = 0;
    int answer = -1 << 60;

    for (int i = 0; i < nums.length; i++) {
      prefix += nums[i];
      int mod = (i + 1) % k;

      answer = (prefix - minPrefix[mod] > answer) ? prefix - minPrefix[mod] : answer;
      minPrefix[mod] = (prefix < minPrefix[mod]) ? prefix : minPrefix[mod];
    }

    return answer;
  }
}

File: NOVEMBER-2025/27-11-2025/3381.kt
class Solution {
    fun maxSubarraySum(nums: IntArray, k: Int): Long {
        val INF = Long.MAX_VALUE
        val minPrefix = LongArray(k) { INF }
        minPrefix[0] = 0L

        var prefix = 0L
        var answer = Long.MIN_VALUE

        for (i in nums.indices) {
            prefix += nums[i]
            val mod = (i + 1) % k

            if (minPrefix[mod] != INF) {
                val candidate = prefix - minPrefix[mod]
                if (candidate > answer) {
                    answer = candidate
                }
            }

            if (prefix < minPrefix[mod]) {
                minPrefix[mod] = prefix
            }
        }

        return answer
    }
}